usr/share/rear/prep/TAPE/default/400_check_tape_drive.sh

```diff
@@ -2,7 +2,7 @@
 REQUIRED_PROGS+=( mt )
 
 # Test for cciss driver and include the necessary tools
-if grep -q '^cciss ' /proc/modules; then
+if grep -q -e '^cciss ' -e '^hpsa' /proc/modules; then
     PROGS+=( "${PROGS_OBDR[@]}" )
```

Comment:
What is the reason that there is no trailing space character
in `'^hpsa'` in contrast to `'^cciss '` that has one?

Reply:
I have a more general question:
Why to make OBDR support conditional on specific drivers (cciss and hpsa)?
Is there a reason why an OBDR-capable tape should not work with any HBA, not just the ones handled by these drivers?
The code was introduced in 3ed01dec19bc0cc62baf2ed6dccb8dc934e2ad58 with the message "OBDR support for HP specific environments (using CCISS)" but unfortunately it does not explain why it should be restricted to HP-specific environments.
@jhoekx @dagwieers  do you happen to know, please?

Reply:
We only had HP equipment to test when we implemented it. We were consulting for HP at the time (2011?) for a customer that needed OBDR on systems that were also doing DRBD.

I was not aware that OBDR was supported by other vendors at all.

There was no information about it back then. It was quite difficult to even figure out how it worked on HP drives as the only software where the source code was available to us that supported it was MondoRescue. We eventually found a manual about the tape drive we were using that documented the SCSI commands we needed.

Reply:
Thank you @jhoekx for the quick reply and congratulations for having found the docs back then.
According to my research, OBDR is indeed HP-specific. But isn't it implemented entirely in the tape drive firmware, making it only HP-drive specific, and not HP-adapter specific? IIUC the tape can pretend that it is a CD-ROM drive, and it should not need anything special in the HBA for this to work.
@maslo64 since you have access to OBDR-capable hardware - what do you think?

Reply:
I'm using mostly HP(E) hardware, so can't test on systems from other vendors, but would agreed that this limitation for HBA is not necessary needed in order to work with any server and HP LTO. If someone will be performing restore from OBDR, he will most likely need tools in REQUIRED_PROGS_OBDR[@].
We were previously using Mondo/Mindi , but switches to ReaR as it works more reliably to our case.

Reply:
IMO this section should then go to a script under usr/share/rear/prep/OBDR/default instead of TAPE (since it will need to be invoked if and only if OBDR is used, right?) and not be conditional on any particular HBA driver.
By the way, `PROGS_OBDR` does not even exist (only `REQUIRED_PROGS_OBDR` does).

Reply:
Unfortunately, I have OBDR capable hardware available only few times per year, and mostly for DR testing, I wanted to perform absolute minimal changes to the code, which worked for me, so any additional testing would would be in far future.